include/seqan3/io/structure_file/output.hpp

```diff
@@ -332,38 +332,70 @@ class structure_file_out
     ~structure_file_out() = default;
 
     /*!\brief Construct from filename.
-     * \param[in] _file_name Path to the file you wish to open.
+     * \param[in] filename Path to the file you wish to open.
      * \param[in] fields_tag A seqan3::fields tag. [optional]
      *
      * \details
      *
      * In addition to the file name, you may specify a custom seqan3::fields type which may be easier than
      * defining all the template parameters.
+     *
+     * ### Compression
+     *
+     * This constructor transparently applies a compression stream on top of the file stream in case
+     * the given file extension suggests the user wants this.
+     * See the section on \link io_compression compression and decompression \endlink for more information.
      */
-    structure_file_out(filesystem::path const & _file_name,
-                       selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{})
+    structure_file_out(filesystem::path filename,
+                       selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{}) :
+        primary_stream{new std::ofstream{filename, std::ios_base::out | std::ios::binary}, stream_deleter_default}
     {
-        // open stream
-        stream.open(_file_name, std::ios_base::out | std::ios::binary);
-        if (!stream.is_open())
+        if (!primary_stream->good())
             throw file_open_error{"Could not open file for writing."};
 
+        // possibly add intermediate compression stream
+        secondary_stream = detail::make_secondary_ostream(*primary_stream, filename);
+
         // initialise format handler or throw if format is not found
-        detail::set_format(format, _file_name);
+        detail::set_format(format, filename);
     }
 
     /*!\brief Construct from an existing stream and with specified format.
      * \tparam file_format The format of the file in the stream, must satisfy
      * seqan3::structure_file_output_format_concept.
-     * \param[in] _stream  The stream to operate on (this must be std::move'd in!).
+     * \param[in,out] stream  The stream to write to, must be derived of std::basic_ostream<stream_char_t>.
```

Comment:
```suggestion
     * \param[in,out] stream  The stream to write to, must be derived of std::basic_ostream.
```
`stream_char_t` is not defined in this function

Reply:
ok!